plugins/modules/ec2_metadata_facts.py

```diff
@@ -484,11 +485,39 @@ def __init__(
         self._token = None
         self._prefix = "ansible_ec2_%s"
 
+    def _decode(self, data):
+        try:
+            return data.decode("utf-8")
+        except UnicodeDecodeError:
+            # Decoding as UTF-8 failed, return data without raising an error
+            self.module.warn("Decoding user-data as UTF-8 failed, return data as is ignoring any error")
+            return data.decode("utf-8", errors="ignore")
+
+    def decode_user_data(self, data):
+        is_compressed = False
+
+        # Check if data is compressed using zlib header
+        if data.startswith(b"\x78\x9c") or data.startswith(b"\x1f\x8b"):
+            is_compressed = True
+
+        if is_compressed:
+            # Data is compressed, attempt decompression and decode using UTF-8
+            try:
+                decompressed = zlib.decompress(data, zlib.MAX_WBITS | 32)
+                return self._decode(decompressed)
+            except zlib.error:
```

Comment:
Not a big fan of this exception being dropped on the floor, at the least we should warn.